vdc/Makefile

```diff
@@ -1,51 +1,108 @@
+export REGION ?= us-central1
+export ZONE ?= us-central1-a
 export DOMAIN ?= staging.hail.is
-export IP ?= 35.188.91.25
+export IP ?= $(shell gcloud --project $(PROJECT) compute addresses describe --region $(REGION) site --format='value(address)')
+
+# run letsencrypt, or take from letsencrypt-config.yaml
+RUN_LETSENCRYPT = 1
 
 build-out: create-deployment build-out-k8s
 
-create-deployment:
+echo-ip:
+	echo IP=$(IP)
+
+activate-deploy:
+	gcloud iam service-accounts keys create deploy-sa-key.json \
+	  --iam-account deploy@$(PROJECT).iam.gserviceaccount.com
+	gcloud auth activate-service-account --key-file=deploy-sa-key.json
+	rm deploy-sa-key.json
+
+gcloud-config:
+	gcloud config set account deploy@$(PROJECT).iam.gserviceaccount.com
 	gcloud config set project $(PROJECT)
-	gcloud beta -q deployment-manager deployments create default --config config.yaml
-# FIXME integrate into config
-	gsutil iam ch serviceAccount:vdc-svc@$(PROJECT).iam.gserviceaccount.com:objectViewer gs://artifacts.$(PROJECT).appspot.com
+	gcloud config set compute/region $(REGION)
+	gcloud config set compute/zone $(ZONE)
 	gcloud container clusters get-credentials vdc
-	kubectl create clusterrolebinding project-editor-cluster-admin-binding --clusterrole cluster-admin --user projectEditor:$(PROJECT)
 
-build-out-k8s: k8s-config create-k8s-secrets create-k8s-certs-volume run-letsencrypt deploy-k8s
+# deployment manager cannot manage appspot.com bucket because it requires domain ownership verification
+create-artifacts-bucket:
+	docker pull alpine:3.8
+	docker tag alpine:3.8 gcr.io/$(PROJECT)/alpine:3.8
+	docker push gcr.io/$(PROJECT)/alpine:3.8
+
+create-deployment: gcloud-config create-artifacts-bucket
+	sed -e 's,@project@,$(PROJECT),g' \
+	  -e 's,@region@,$(REGION),g' \
+	  -e 's,@zone@,$(ZONE),g' \
+	  < gcp-config.yaml.in > gcp-config.yaml
+	gcloud beta -q deployment-manager deployments create default --config gcp-config.yaml
+	kubectl create clusterrolebinding deploy-cluster-admin-binding --clusterrole cluster-admin --user deploy@$(PROJECT).iam.gserviceaccount.com
+# FIXME still trying to figure out out to add this to config
+	gsutil iam ch serviceAccount:gcr-push@$(PROJECT).iam.gserviceaccount.com:admin gs://artifacts.$(PROJECT).appspot.com
+	gsutil iam ch serviceAccount:gcr-pull@$(PROJECT).iam.gserviceaccount.com:objectViewer gs://artifacts.$(PROJECT).appspot.com
+	gsutil iam ch serviceAccount:vdc-sa@$(PROJECT).iam.gserviceaccount.com:objectViewer gs://artifacts.$(PROJECT).appspot.com
+
+build-out-k8s: k8s-config create-k8s-secrets run-letsencrypt deploy-k8s
 
 k8s-config:
 	kubectl apply -f k8s-config.yaml
 
 create-k8s-secrets:
 	kubectl apply -f secrets.yaml
-	kubectl create secret generic upload-oauth2-client-secret \
-	  --from-file=./client_secret.json
-
-create-k8s-certs-volume:
-	kubectl apply -f ../site/certs-volume.yaml
+	kubectl delete secrets  --ignore-not-found=true gcr-push-service-account-key gcr-pull-key letsencrypt-config
```

Comment:
> committing to no changes that aren't in the config

I'm not sure what this means.

There should be two kinds of resources: managed and unmanaged.  This requires some work, managed resources need to be appropriately labeled and update commands have specify the full set of resources at once to know what running resources have been deleted using `kubectl apply --prune`.

The reason I'm deleting here is because I call "kubectl create" below in this rule, and that fails if the resource exists (unlike the apply just below).

In the long term, secrets.yaml should be combined with the pre-configured secrets and placed with all the k8s resources to apply --prune.